src/full_node/block_header_validation.py

```diff
@@ -36,26 +36,64 @@ async def validate_unfinished_header_block(
     height_to_hash: Dict[uint32, bytes32],
     header_block: UnfinishedHeaderBlock,
     check_filter: bool,
+    skip_overflow_last_ss_validation: bool = False,
 ) -> Tuple[Optional[uint64], Optional[ValidationError]]:
     """
     Validates an unfinished header block. This is a block without the infusion VDFs (unfinished)
     and without transactions and transaction info (header). Returns (required_iters, error).
+    TODO: handle unfinished overflow block with no end of slot
     """
     # 1. Check that the previous block exists in the blockchain, or that it is correct
     new_sub_slot: bool = len(header_block.finished_sub_slots) > 0
     prev_sb = sub_blocks.get(header_block.prev_header_hash, None)
     genesis_block = prev_sb is None
-    height: uint32 = uint32(0)
+
     if genesis_block and header_block.prev_header_hash != constants.GENESIS_PREV_HASH:
         return None, ValidationError(Err.INVALID_PREV_BLOCK_HASH)
 
-    if not genesis_block:
+    if genesis_block:
+        height: uint32 = uint32(0)
+        ips = constants.IPS_STARTING
+        difficulty = constants.DIFFICULTY_STARTING
+        finishes_se: bool = False
+        finishes_epoch: bool = False
+    else:
         height: uint32 = uint32(prev_sb.height + 1)
+        finishes_se: bool = finishes_sub_epoch(
+            constants, prev_sb.height, prev_sb.deficit, False, sub_blocks, prev_sb.prev_hash
+        )
+        finishes_epoch: bool = finishes_sub_epoch(
+            constants, prev_sb.height, prev_sb.deficit, True, sub_blocks, prev_sb.prev_hash
+        )
 
-    finishes_se = finishes_sub_epoch(constants, prev_sb, False, sub_blocks)
-    finishes_epoch: bool = finishes_sub_epoch(constants, prev_sb, True, sub_blocks)
+        ips, difficulty = get_ips_and_difficulty(constants, header_block, height_to_hash, prev_sb, sub_blocks)
+
+    # 4. Check proof of space
```

Comment:
@mariano54 
1. any special reason for validating 4 first ? 
2. i think we should validate all correct fields exist first and order the rest of validation steps 
from easiest to validate to the hardest

Reply:
To validate step 2 (the finished slots), we need to know whether or not we are overflow. This is because when validating an unfinished overflow block we do not have the "finished slots" that are necessary to verify step 2, so this has to be skipped.

To know whether it's overflow or not, and for that, we need to know the required_iters (which requires veifying the proof of space).

Reply:
ok, but i don't see us using "overflow" until after after finished slots validation (line 361 is next usage) , are we missing some code in 2 ?

what do you mean "has to be skipped", we will only be able to validate this when the overflow block is finished  right ?